script/echoserver.py

```diff
@@ -0,0 +1,54 @@
+import socket
+import sys
+
+servport = 1234
+proxyport = 1111
+
+def server(port): 
+    print('starting server on port {}'.format(port))
+    address = ('', port)
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+    # allow reuse of socket addresses for faster testing
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+    sock.setsockopt(socket.SOL_IP, socket.IP_TRANSPARENT, 1)
```

Comment:
Proxy server needs this to extract the original connection destination instead of its own address